### PR TITLE
Simplify project structure & move to .NET Standard

### PR DESCRIPTION
**Pull request by onovotny**
@kpanwar As discussed over email, this is the general idea. The project (src\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj) currently builds all outputs.

You need to use VS 2017 RC2 for this. You can build in either VS or the command line with `msbuild /t:restore;build`

Or to build a nuget package
`msbuild /t:restore;pack`

There's still a few housekeeping things left to do, but these are easy:
1. [x] Attach version number
2. [x] Attach strong name signing 
3. [x] Re-attach test projects
4. [ ] Re-enable code analysis
5. [ ] Re-Attach Sample projects

The bulk of the file changes are renames; no content has been changed in the source files (yet...). The existing csproj have been removed and a new `Microsoft.Identity.Client.csproj` now contains all of the source.
MSAL issue tracking this feature -- https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/130

### CODE REVIEW

File: src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    <Reference Include="System.Data" />
+    <Reference Include="System.Xml" />
+    <Compile Include="Platforms\net45\*.cs" />
+    <Compile Include="Platforms\Common\MsalEventSource.cs" />
+    <Compile Include="Platforms\Common\PublicClientApplicationExtensions.cs" />
+    <Compile Include="Platforms\Common\ClientAssertionCertificate.cs" />
+    <Compile Include="Platforms\Common\Logger.cs" />
+    <Compile Include="Platforms\Common\MsalOption.cs" />    
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'wpa81' or '$(TargetFramework)' == 'win81' or '$(TargetFramework)' == 'uap10.0' ">
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: we don't plan on supporting wpa81. I think should not have defined in the csproj.

File: src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 2]
```diff
@@ -0,0 +1,144 @@
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+  <PropertyGroup>
+    <TargetFrameworks>netstandard1.2;netstandard1.3;net45;win81;wpa81;MonoAndroid70;Xamarin.iOS10</TargetFrameworks>
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: wpa81 is not going to be supported.

**Comment by kpanwar** [comment]
@onovotny - thanks for the PR! One more item to add in your list would be to "Re-Attach Sample projects".

File: src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj (hunk 2 above)
Comment (onovotny) [inline src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: Do you plan on supporting `win81` in this release? If so, `wpa81` is identical; no changes at all.

**Comment by onovotny** [comment]
@kpanwar yep, will update the list. I wanted to make sure we were on the same page with the concept before diving into the weeds to tie things off. If you're on-board then I'll definitely proceed.

Thanks

**Comment by onovotny** [comment]
@kpanwar The `sdl7.0.ruleset` isn't available in the repo. Could you please check it in?

**Comment by onovotny** [comment]
@kpanwar I've gone as far as I can for now until a few more bugs are fixed in the toolchain.

These issues are currently blocking me from testing out the samples.
https://github.com/dotnet/sdk/issues/535
https://github.com/dotnet/roslyn-project-system/issues/936

The unit test project does work and all tests pass.

**Comment by onovotny** [comment]
I've updated the PR off of the latest `dev` but with one update -- it now pulls in the files from the existing directories to avoid moving them until the PR is ready to move into dev. Otherwise there are too many merge conflicts.

When this PR is ready to be merged, the files should be moved one last time (or it could be a separate PR that gets quickly merged).

**Comment by kpanwar** [comment]
@onovotny - I do see activity on the issues you linked, but do you know if there is a known release date for the fixes?

**Pull request merged**